packages/cli/src/utils/run-all-tests.utils.ts

```diff
@@ -5,6 +5,25 @@ import log from "loglevel";
 import { getLatestTestRunResults } from "../api/test-run.api";
 import { DetailedTestCaseResult } from "../config/config.types";
 
+export const mergeTestCases = (
+  ...testSuites: (TestCase[] | null | undefined)[]
+): TestCase[] => {
+  const seenSessionIds = new Set<string>();
+
+  return testSuites.flatMap((testSuite) => {
+    if (testSuite == null) {
+      return [];
+    }
+    return testSuite.flatMap((testCase) => {
+      if (seenSessionIds.has(testCase.sessionId)) {
+        return [];
+      }
+      seenSessionIds.add(testCase.sessionId);
+      return [testCase];
+    });
```

Comment:
Note if we want the repeat stuff that the main meticulous repo does to still work then we may need to do:

```ts
    const novelTestCases = testSuite.filter((testCase) => !seenSessionIds.has(testCase.sessionId));
    testSuite.forEach((testCase) => seenSessionIds.add(testCase.sessionId));
    return novelTestCases;
```

So that it only dedups across testSuites and not within testSuites.